### PR TITLE
Node e2e: switch most of the tests to use the e2e framework

### PR DESCRIPTION
**Pull request by yujuhong**
This commit switches the tests to use the e2e framework, so that namespace
creation/deletion is handled between tests.

### CODE REVIEW

File: test/e2e_node/mirror_pod_test.go [hunk 1]
```diff
@@ -24,25 +24,21 @@ import (
...
-		// Setup the apiserver client
-		cl = client.NewOrDie(&restclient.Config{Host: *apiServerAddress})
-	})
-	ns := "mirror-pod"
+var _ = framework.KubeDescribe("MirrorPod", func() {
+	f := NewDefaultFramework("mirror-pod")
 	Context("when create a mirror pod ", func() {
 		var staticPodName, mirrorPodName string
 		BeforeEach(func() {
+			ns := f.Namespace.Name
```
Comment (Random-Liu) [inline test/e2e_node/mirror_pod_test.go]: Just use a podClient (`client.Pods(ns)`) is cleaner. I'll clean up this in the future, mark here. :)

Comment (yujuhong) [inline test/e2e_node/mirror_pod_test.go]: We need utility functions for creating pods in general in the node e2e suite. This PR doesn't do that, unfortunately.

**Comment by Random-Liu** [comment]
LGTM.

**Comment by Random-Liu** [comment]
@k8s-bot test this issue #27290

**Comment by yujuhong** [comment]
@Random-Liu  I think you merge this commit in your PR already. Closing.

**Comment by Random-Liu** [comment]
@yujuhong ?? I didn't expect that to be merged so quickly. Sorry, :P

**Comment by yujuhong** [comment]
Haha it's ok. I got confused when trying to rebase, and then discovered that the commit was already there.

**Pull request closed without merging**